Add c64dtv cpu support

brazso opened the pull request:
Please consider my changes to enhance cc65 with **c64dtv extra op codes** support. I wrote a simple [hello-world project](https://github.com/brazso/c64dtv-hello-cc65/tree/develop) to test the new options of cc65.
In cc65 there is a new cpu option named **6502dtv**. It contains the op codes of 6502x extended with dtv extra op codes. The disassembler (da65) was also modified to detect and identify these new op codes.

brazso commented:
Please add **enhancement** label to this pull request. It seems I have no permission to do it...

oliverschmidt commented on src/ca65/instr.c:
```diff
@@ -303,6 +303,97 @@ static const struct {
+    {
+        { "ADC",  0x080A26C, 0x60, 0, PutAll },
+        { "ALR",  0x0800000, 0x4B, 0, PutAll },         /* X */
+        { "ANC",  0x0800000, 0x0B, 0, PutAll },         /* X -DTV? */
```
I guess the comment wants  to read `X, -DTV` (with comma) here too...

brazso replied:
You are right, I have corrected the referred mistyped comment.

oliverschmidt commented on src/ca65/instr.c:
```diff
@@ -303,6 +303,97 @@ static const struct {
 
+/* Instruction table for the 6502 with illegal instructions (X) and DTV 
+** extra opcodes (DTV). Some illegal instructions (X, -DTV?) might be not 
+** supported by DTV. They should be tested on the DTV hardware. 
```
Do you have a DTV hardware? Can you conduct the tests?

mrdudz replied:
It's fairly accurate in VICE *afaik*. Biggest trap is that some of these do not work 100% the same as on 6502/10 iirc.

brazso replied:
I have 2 devices (DTV2: 050917 and 050927) however none of them is modded yet sot I cannot upload any programs to them. I think running a illegal instructions test on VICE emulator does not make sense, I do not know how accurate it is. Unfortunately the [related description](https://www.c64-wiki.com/wiki/C64DTV_Programming_Guide#C64_emulation_issues_.28for_the_DTV2.2F3.29) in the programming guide is also rather vague and insufficient. We can remove all illegals codes from my new InsTab6502DTV table in cc65 or they are tagged in comments that they should be verified later on the hardware.

oliverschmidt reviewed (changes requested):
I'm in general open to merging this PR. However, please see my comments.

greg-king5 commented:
`6502dtv` should be added to the list of supported CPUs in `"doc/da65.sgml"`.

mrdudz commented:
> I think running a illegal instructions test on VICE emulator does not make sense, I do not know how accurate it is.

certainly accurate enough to check which illegals exist and which do not - TLR and Noojopa tested this quite a bit when they made the emulator. you could check the dtv port of the lorenz testbench for a list: https://sourceforge.net/p/vice-emu/code/HEAD/tree/testprogs/DTV/tsuitDTV/

brazso commented:
> da65.sgml

I have updated da65 with the new cpu option setting named 6502dtv

brazso commented:
> certainly accurate enough to check which illegals exist and which do not - TLR and Noojopa tested this quite a bit when they made the emulator. you could check the dtv port of the lorenz testbench for a list: https://sourceforge.net/p/vice-emu/code/HEAD/tree/testprogs/DTV/tsuitDTV/

I checked the mentioned test suites, the opcodes verified by VICE-emu, and[ synchronized them](https://github.com/brazso/cc65/commit/59ae1a559950819cc9245b1dcaafe9674f34a566) with the instructions table for 6502dtv.

oliverschmidt commented:
So from my POV this PR is ready to be merged. Any Objections?

greg-king5 commented:
@brazso, what about da65's table?  Do you plan to synchronize it?

brazso reviewed (commented):
> @brazso, what about da65's table? Do you plan to synchronize it?
Thanks for your note. I have updated the table. Having removed instructions from ca65's table (synchronized with VICE-emu tests) got illegal flags in da65's table.

oliverschmidt commented:
As far as I understand @greg-king5's comment has been taken care of -> merging...

oliverschmidt commented:
@brazso: Unfortunately it did go unnoticed that this PR breaks our CI :-(

```
asm/4510-cpudetect.bin
Makefile:71: recipe for target '../../testwrk/asm/4510-cpudetect.bin' failed
make[2]: *** [../../testwrk/asm/4510-cpudetect.bin] Error 1
```

The source for these tests is https://github.com/cc65/cc65/tree/master/test/asm. Can you have a look what goes wrong and ideally provide a PR fixing it? Maybe you even want to extend the tests for the newly supported CPU...

Thanks in advance, Oliver

oliverschmidt commented:
I fixed the regression caused by this PR myself with https://github.com/cc65/cc65/commit/69c0363c7e0be4a230f00616da69ea9fdcf80c0b.

However, it might still be desirable to extend https://github.com/cc65/cc65/tree/master/test/asm for the newly supported CPU - as we just learned the hard way those tests aren't exactly useless ;-)

brazso commented:
@oliverschmidt: Thanks for the fix, you were quicker than me. I had to create a new [repo](https://github.com/brazso/cc65/tree/c64dtv), because after merging my pull request to the cc65's master branch, I had deleted the previous one.
In that new macros were added for 6502DTV cpu and those are already used in test/asm. Moreover I have removed 6502X as a subset of the new 6502DTV because some of them are not supported on DTV in fact. The latter one is only a superset of 6502. Do I create a new pull request? Can we run CI as a check without merging my fork to cc65:master?

oliverschmidt commented:
Thanks for the background info :-)

* Yes, please create a new pull request. As soon as a pull request is merged it's "done".

* The CI automatically runs for every pull request (and additional push into the pull request branch). However, currently there's an issue with automatic feedback of CI in the pull request page. You/me/we have currently to explicitly check the CI for the pull request at https://travis-ci.org/github/cc65/cc65/p ull_requests
